### DISCUSSION TITLE
Does perception work with NSManagedObjects?

### DISCUSSION THREAD

**Discussion by kdo1234** [pointfreeco/swift-perception#89] [Q&A] [answered]
Does perception work with NSManagedObjects? Thank you

**Comment by mbrandonw** [comment, marked as answer]
Hi @kdo1234, the Perception library does not add any additional functionality that Swift's native Observation framework doesn't have. So, if you can use `@Observable` with `NSManagedObject` subclasses, then you can with `@Perceptible` too.

But I'm not entirely sure you can use `@Observable` with `NSManagedObject` subclasses. You should give it a shot and see what goes wrong. If you have questions about that it may be better suited for Apple's forums.
